Sources/BraintreeApplePay/BTApplePayClient.swift

```diff
@@ -0,0 +1,129 @@
+import Foundation
+import PassKit
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+/// Used to process Apple Pay payments
+@objcMembers public class BTApplePayClient: NSObject {
+
+    //MARK: - Internal Properties
+
+    /// Exposed for testing to get the instance of BTAPIClient
+    var apiClient: BTAPIClient
+
+    // MARK: - Initializer
+
+    /// Creates an Apple Pay client
+    /// - Parameter apiClient: An API client
+    @objc(initWithAPIClient:)
+    public init(apiClient: BTAPIClient) {
+        self.apiClient = apiClient
+    }
+
+    // MARK: - Public Methods
+
+    @objc(paymentRequest:)
+    public func paymentRequest(completion: @escaping (PKPaymentRequest?, Error?) -> Void) {
+        apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
+            if let error {
+                self.apiClient.sendAnalyticsEvent("ios.apple-pay.error.configuration")
+                self.completionHandler(onMainThreadWithPaymentRequest: nil, error: error, completion: completion)
+                return
+            }
+
+            guard let configuration, configuration.isApplePayEnabled else {
+                self.apiClient.sendAnalyticsEvent("ios.apple-pay.error.disabled")
+                self.completionHandler(onMainThreadWithPaymentRequest: nil, error: BTApplePayError.unsupported, completion: completion)
+                return
+            }
+
+            let paymentRequest = PKPaymentRequest()
+            paymentRequest.countryCode = configuration.applePayCountryCode ?? ""
+            paymentRequest.currencyCode = configuration.applePayCurrencyCode ?? ""
+            paymentRequest.merchantIdentifier = configuration.applePayMerchantIdentifier ?? ""
+            paymentRequest.supportedNetworks = configuration.applePaySupportedNetworks ?? []
+
+            self.completionHandler(onMainThreadWithPaymentRequest: paymentRequest, error: nil, completion: completion)
+        }
+    }
+
+    @objc(tokenizeApplePayPayment:completion:)
+    public func tokenize(_ payment: PKPayment, completion: @escaping (BTApplePayCardNonce?, Error?) -> Void) {
+        apiClient.sendAnalyticsEvent("ios.apple-pay.start")
+
+        apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
+            if let error {
+                self.apiClient.sendAnalyticsEvent("ios.apple-pay.error.configuration")
+                completion(nil, error)
+                return
+            }
+
+            if let configuration,
+               configuration.json?["applePay"]["status"].isString == false ||
+               configuration.json?["applePay"]["status"].asString() == "off" {
```

Comment:
Can we re-use [this logic](https://github.com/braintree/braintree_ios/blob/dd0cbeae0b48606885ed3684b7463ad799d8ee13/Sources/BraintreeCore/BTConfiguration%2BExtensions.swift#L57-L60) for this check?

Reply:
Yep, will update this. I am not sure why it was duplicated in Obj-C in the [configuration here](https://github.com/braintree/braintree_ios/blob/230d8f0fd7dbdfc3fd9d9230e9472f3b705e3669/Sources/BraintreeApplePay/BTConfiguration%2BApplePay.m#L9-L12) and the [implementation here](https://github.com/braintree/braintree_ios/blob/230d8f0fd7dbdfc3fd9d9230e9472f3b705e3669/Sources/BraintreeApplePay/BTApplePayClient.m#L115-L116). Will update this as well as move the Apple Pay configuration extension into the Apple Pay module as we have with the other Swift modules.